drivers: nrf_clock_calibration: fix errata 192 workaround

xudongzheng opened the pull request:
Workaround for errata 192 should only apply on nRF52810, nRF52832, and
nRF52840.

Fixes #43930

nordic-krch commented on drivers/clock_control/nrf_clock_calibration.c:
```diff
@@ -108,7 +108,11 @@ static void cal_lf_callback(struct onoff_manager *mgr,
 {
 	/* Workaround for Errata 192 */
-	if (IS_ENABLED(CONFIG_SOC_SERIES_NRF52X)) {
+	if (IS_ENABLED(CONFIG_SOC_NRF52810_QFAA) ||
```
well spotted but actually you could just remove whole if. Code is using nrfx driver and workaround is applied there:
https://github.com/zephyrproject-rtos/hal_nordic/blob/1f9145e8c8359f103b4ac5085c6f5d2e66a2557d/nrfx/drivers/src/nrfx_clock.c#L415

nordic-krch reviewed (changes requested):
Please add sign-off to the commit message.
Also start commit message with:
`drivers: clock_control: nrf_clock_calibration:...`

xudongzheng committed:
drivers: clock_control: nrf_clock_calibration: remove errata workaround

Workaround for errata 192 is unnecessary as it is applied within
nrfx_clock_calibration_start().

Fixes #43930

Signed-off-by: Xudong Zheng <7pkvm5aw@slicealias.com>
